pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,375 @@
+"""Module for classification function."""
+import numpy as np
+
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.model_selection import train_test_split
+
+from qiskit import BasicAer, IBMQ
+from qiskit.circuit.library import ZZFeatureMap, TwoLocal
+from qiskit.aqua import QuantumInstance, aqua_globals
+from qiskit.aqua.quantum_instance import logger
+from qiskit.aqua.algorithms import QSVM, SklearnSVM, VQC
+from qiskit.aqua.utils import get_feature_dimension
+from qiskit.providers.ibmq import least_busy
+from qiskit.aqua.components.optimizers import SPSA
+from datetime import datetime
+import logging
+logger.level = logging.INFO
+
+
+class QuanticClassifierBase(BaseEstimator, ClassifierMixin):
+
+    """Quantum classification.
+
+    This class implements a SKLearn wrapper around Qiskit library.
+    It provides a mean to run classification tasks on a local and
+    simulated quantum computer or a remote and real quantum computer.
+    Difference between simulated and real quantum computer will be that:
+
+    * there is no noise on a simulated quantum computer
+      (so results are better);
+    * real quantum computer are quicker than simulator;
+    * real quantum computer tasks are assigned to a queue
+      before being executed on a back-end.
+
+    WARNING: At the moment this implementation only supports binary
+    classification (eg. 3d vs Non-vectorize experiment)
+
+    Parameters
+    ----------
+    labels : (int, int)
+        Label of the first and second classes
+    quantum : bool (default: True)
+        - If true will run on local or remote backend
+        (depending on q_account_token value).
+        - If false, will perform classical computing instead
+    q_account_token : string (default:None)
+        If quantum==True and q_account_token provided,
+        the classification task will be running on a IBM quantum backend
+    process_vector : callable (default: lambda v: v)
+        Additional processing on the input vectors. eg: downsampling
+    verbose : bool (default:True)
+        If true will output all intermediate results and logs
+    test_input : dict (default: {}})
+        Contains vectorized test set for the two classes
+    test_per : double (default: 0.33)
+        Indicate the percent of the fitting set that should be kept for
+        testing. The remaining percent are used for training.
+        This is due to the design of quantum classifiers which require
+        training, testing and predicting sets.
```

Comment:
A predicting set is another testing set, likely belonging to another dataset. Such as in the example you provide me, where the mesh data that are passed to the `predict_proba` methods do not belong to the initial dataset.
![image](https://user-images.githubusercontent.com/6229031/135981115-444952bf-8e20-4755-9efd-db4868ad180c.png)

Reply:
I don't understand what is a prediction set, because in the toy example, `xx` and `yy` are simply processed like another test set. In classical ML, training is independent from testing; and a fitted classifier can be used on multiple different test sets without problem.
Is prediction set a quantum ML specificity?